tests/application/job_service/test_job_service.py

```diff
@@ -0,0 +1,154 @@
+from datetime import datetime
+from typing import Type
+from unittest.mock import Mock, MagicMock, call, patch
+
+import pytest
+from freezegun import freeze_time
+from opyoid import Module, Injector
+from sqlalchemy.orm import Session
+
+from erica.application.FreischaltCode.FreischaltCode import BaseDto
+from erica.application.JobService.job_service import JobService, JobServiceInterface
+from erica.domain.BackgroundJobs.BackgroundJobInterface import BackgroundJobInterface
+from erica.domain.EricaAuftrag.EricaAuftrag import EricaAuftrag
+from erica.domain.Repositories.EricaAuftragRepositoryInterface import EricaAuftragRepositoryInterface
+from erica.domain.Shared.EricaAuftrag import AuftragType
+from erica.erica_legacy.request_processing.requests_controller import CheckTaxNumberRequestController, \
+    EricaRequestController
+from erica.infrastructure.sqlalchemy.repositories.EricaAuftragRepository import EricaAuftragRepository
+
+
+class MockEricaRequestRepository(EricaAuftragRepository, list):
+
+    def __init__(self, db_connection: Session = None):
+        self.DatabaseEntity = None
+        self.DomainModel = None
+
+    def create(self, model):
+        self.append(model)
+        model.id = "1234"
+        return model
+
+    def get(self, skip: int = 0, limit: int = 100):
+        return self
+
+    def get_by_id(self, entity_id):
+        return [entity for entity in self if entity.id == entity_id][0]
+
+    def update(self, model_id, model):
+        for entity, index in enumerate(self):
+            if entity.id == model_id:
+                self[index] = model
+        return model
+
+    def delete(self, entity_id):
+        for entity, index in enumerate(self):
+            if entity.id == entity_id:
+                self.pop(index)
+
+
+class MockRequestController(CheckTaxNumberRequestController):
+    call_list = []
+
+    @classmethod
+    def process(cls, *args, **kwargs):
+        cls.call_list.append({'args': [*args], 'kwargs': {**kwargs}})
+
+
+class PickableMock(Mock):
+
+    def __reduce__(self):
+        return Mock, ()
+
+
+class MockDto(BaseDto):
+    name: str
+    friend: str
+
+
+class MockBackgroundJob(BackgroundJobInterface, MagicMock):
+    pass
+
+
+class TestJobServiceQueue:
+
+    @freeze_time("2001-01-03 08:22:00")
+    @pytest.mark.freeze_uuids
+    def test_if_input_data_provided_then_add_request_to_repository(self):
+        mock_job = PickableMock()
+        service = JobService(job_repository=MockEricaRequestRepository(), background_worker=MagicMock(),
+                             request_controller=MockRequestController, payload_type=MockDto, job_method=mock_job)
+        input_data = MockDto.parse_obj({'name': 'Batman', 'friend': 'Joker'})
+
+        service.apply_queued_to_elster(input_data, job_type=AuftragType.freischalt_code_activate)
```

Comment:
Hi! Just a quick note - I really misunderstood this method. `apply_queued_to_elster` sounds like executing the job directly. And this is write the opposite of what is being done here - merely queuing the job. Therefore, I'd very much prefer a renaming to `add_to_queue` `add_to_elster_job_queue` `add_as_elster_job_to queue` or something along the lines of this.

Reply:
then `add_to_queue`